pandas/core/common.py

```diff
@@ -1403,6 +1403,10 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                if "floordiv" in name:  # (PR 9308)
+                    nan_mask = ((y == 0) & (x == 0)).ravel()
+                    np.putmask(result, nan_mask, np.nan)
+
```

Comment:
So my comparision is direct numpy, e.g. what is the overhead that pandas is adding. Here is a 100MM frame (I constructed same as you). Note that using ``.values`` like this works (with no copy) only if its only a  single dtype (as is the case here). So yes I think a 3x slowdown is too much. (granted we are doing more/better ops, but should be only a limited overhead).

```
In [65]: xx.size
Out[65]: 100000000

In [66]: xx.size/1e6
Out[66]: 100.0

In [67]: %timeit xx.values*2
1 loops, best of 3: 521 ms per loop

In [68]: %timeit xx.values/0
1 loops, best of 3: 613 ms per loop
```